d2l-search-facets-grouping.html

```diff
@@ -0,0 +1,135 @@
+	<template strip-whitespace>
+		<style>
+			:host {
+				display: flex;
+				flex-direction: column;
+			}
+			.d2l-search-facets-grouping-label {
+				@apply --d2l-body-compact-text;
+				font-weight: bold;
+			}
+
+			.d2l-search-facets-grouping-show-more > button {
+				@apply --d2l-body-compact-text;
+				border: none;
+				color: var(--d2l-color-celestine);
+				font-family: inherit;
+				outline: none;
+				padding: 0px;
+			}
+
+			.d2l-search-facets-grouping-show-more > button:hover,
+			.d2l-search-facets-grouping-show-more > button:focus {
+				cursor: pointer;
+				font-weight: bold;
+			}
+
+		</style>
+		<span class="d2l-search-facets-grouping-label">[[label]]</span>
+		<d2l-input-checkbox-spacer></d2l-input-checkbox-spacer>
+		<slot></slot>
+		<template is="dom-if" if="{{_showMoreButtonVisible}}">
+			<d2l-input-checkbox-spacer class="d2l-search-facets-grouping-show-more">
+				<button on-click="_showMore">More</button>
+			</d2l-input-checkbox-spacer>
+		</template>
+	</template>
+	<script>
+		/**
+		 * `<d2l-search-facets-grouping>`
+		 * A grouping of search facet options that keeps tracks of selected options
+		 */
+		class SearchFacetsGrouping extends Polymer.Element {
+			static get is() { return 'd2l-search-facets-grouping'; }
+			static get properties() {
+				return {
+					/**
+					* List of <d2l-search-facets-option> elements
+					*/
+					_optionElements: {
+						type: Array,
+						value: function() { return []; },
+					},
+					/**
+					* List of selected options (e.g., ['optionA', ...]s)
+					*/
+					_optionsSelected: {
+						type: Array,
+						value: function() { return []; },
+					},
+					/**
+					* Visibility of 'More' button - true if initiallyShown < _optionElements.length
+					*/
+					_showMoreButtonVisible: {
+						type: Boolean,
+						value: false,
+					},
+					/**
+					* Label of the grouping
+					*/
+					label: {
+						type: String,
+						value: '',
+					},
+					/**
+					* Number of facet options initially displayed. All options are shown by default
+					*/
+					initiallyShown: {
+						type: Number,
+					}
+				};
+			}
+
+			connectedCallback() {
+				super.connectedCallback();
+				this.addEventListener('d2l-search-facets-option-change', function(e) {
+					e.detail.checked
+						? this._optionsSelected.push(e.detail.option)
+						: this._optionsSelected.splice(this._optionsSelected.indexOf(e.detail.option), 1);
+
+					this.dispatchEvent(new CustomEvent(
+						'd2l-search-facets-grouping-change', {
+							bubbles: true, composed: true, detail:
+							{
+								grouping: this.label,
+								selected: this._optionsSelected,
+							}
+						}
+					));
+				}.bind(this));
+
+				// Child elements may not be rendered if this is a <dom-repeat>. Wait for next
+				// render to query for them
+				Polymer.RenderStatus.afterNextRender(this, function() {
+					this._optionElements = Polymer.FlattenedNodesObserver.getFlattenedNodes(this)
```

Comment:
It looks like we end up with 2 sources of truth here - the DOM, and this property.  If additional options are added or removed after this code runs, they would not be reflected in this array.  In addition, if options are removed, this component would still be holding references to those DOM nodes.  I think it is better to treat the DOM as the single source of truth.  But I could be misunderstanding how this is used as well... ?